src/utils/ExportData.js

```diff
@@ -17,7 +19,45 @@ const setUpXml = () => {
   return xml;
 };
 
-const generateKeys = (graph, graphML, elements, type, excludeList) => {
+const getVariableInfo = (variableRegistry, type, element, key) => (
+  variableRegistry[type] && variableRegistry[type][element.type] &&
+  variableRegistry[type][element.type].variables &&
+  variableRegistry[type][element.type].variables[key]
+);
+
+const isVariableRegistryExists = (variableRegistry, type, element, key) => {
+  const knownTypes = ['boolean', 'text', 'number', 'categorical', 'ordinal', 'checkbox',
```

Comment:
I think we need to be careful with this list. I realize a lot of these 'types' were present previously, but the list needs to be revised now that we have a new system that allows us to differentiate between variable types and input components. So, specifically:

- checkbox is an input type - should be removed. It is covered by the boolean type
- enumerable is another word for ordinal
- options is another word for categorical (at least, they refer to the same thing)

Reply:
I can update the existing protocols. So a "checkbox" variable that isn't in the registry is just handled as a string? Or should the registry be updated to have the "boolean" type for that variable? Enumerable and options were lingering in the development protocol, but I can update those.

Reply:
In the case of the factory protocols, I've updated them in a forthcoming PR addressing inputs. I have also created the following wiki page which attempts to map the basic variable types and input components that cover them: https://github.com/codaco/Network-Canvas/wiki/Variable-Types

Reply:
If a variable is encountered that doesn't have a corresponding definition in the variable registry, I think that should probably be a hard failure, since it indicates an invalid protocol. When we are closer to a final release we might be able to relax that, with a fallback of just treating it as text.